src/transaction.cpp

```diff
@@ -816,10 +816,11 @@ void wsrep::transaction::after_applying()
 }
 
 bool wsrep::transaction::bf_abort(
-    wsrep::unique_lock<wsrep::mutex>& lock WSREP_UNUSED,
+    wsrep::unique_lock<wsrep::mutex>& lock,
     wsrep::seqno bf_seqno)
 {
     bool ret(false);
+    enum wsrep::transaction::state prev_state(state());
     assert(lock.owns_lock());
```

Comment:
both state() and prev_state are used in following code.
To avoid confusion, I suggest changing e.g. prev_state -> state_at_enter, and only using state_at_enter after this point